.github/workflows/release.yml

```diff
@@ -12,20 +12,42 @@ jobs:
     steps:
       - uses: actions/checkout@v3
         with:
+          path: ${{ github.ref_name }}
           submodules: true
 
-      - name: Create archives
+      - name: Cleanup
+        shell: bash
+        run: |
+          # Remove all .git folders & files as well as .github folders and .git, .gitattribute etc. files
+          find ${{ github.ref_name }} -name ".git*" -prune -exec rm -rf {} +
+          # Remove CI configs
+          find ${{ github.ref_name }} \( \
+            -name "appveyor.yml" -o \
+            -name ".appveyor.yml" -o \
+            -name "azure-pipelines.yml" -o \
+            -name ".azure-pipelines.yml" -o \
+            -name "circle.yml" -o \
+            -name ".cirrus.yml" -o \
+            -name "codecov.yml" -o \
+            -name ".codecov.yml" -o \
+            -name ".coveralls.yml" -o \
+            -name ".drone.star" -o \
+            -name ".travis.yml" -o \
+            -false \) -delete
+            
+      - name: Create Linux Archives
         run: |
-          cd ..
-          cp -a ${{ github.event.repository.name }} ${{ github.ref_name }}
-          rm -rf ${{ github.ref_name }}/.git
-          zip -q ${{ github.ref_name }}.zip -r ${{ github.ref_name }}
           tar -czf ${{ github.ref_name }}.tar.gz ${{ github.ref_name }}
           tar -cJf ${{ github.ref_name }}.tar.xz ${{ github.ref_name }}
+      - name: Create Windows archives
+        run: |
+          sudo apt-get -o Acquire::Retries=5 install -y dos2unix
+          find ${{ github.ref_name }} -type f -exec unix2dos {} \;
```

Comment:
I think this processes excess files. `.sh` files must retain UNIX line endings even on Windows, and you should not process any binary files, e.g. images in docs.

I still think the best way to do it is to build Windows archives on Windows. git will do the conversion on checkout for you.

Reply:
This certainly sounds way too dangerous.

Reply:
unix2dos in recent version doesn't modify binary files as I've learned. But true on sh files, so added a windows job which makes it a bit more complicated and I hope the release uploader correctly finds the files and combines them into that one release